ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -573,21 +540,19 @@ public static Boolean shouldUpdateValidatorRegistry(BeaconState state)
       throws EpochProcessingException {
     try {
       Boolean check1 = false;
-      UnsignedLong finalized_epoch = state.getFinalized_epoch();
-      UnsignedLong validator_registry_update_epoch = state.getValidator_registry_update_epoch();
-      check1 = finalized_epoch.compareTo(validator_registry_update_epoch) > 0;
+      long finalized_epoch = state.getFinalized_epoch();
+      long validator_registry_update_epoch = state.getValidator_registry_update_epoch();
+      check1 = finalized_epoch > validator_registry_update_epoch;
 
       Boolean check2 = false;
-      UnsignedLong committee_count = BeaconStateUtil.get_current_epoch_committee_count(state);
+      long committee_count = BeaconStateUtil.get_current_epoch_committee_count(state);
       List<Integer> comnmittee_range =
-          IntStream.range(0, committee_count.intValue()).boxed().collect(Collectors.toList());
-      UnsignedLong SHARD_COUNT = UnsignedLong.valueOf(Constants.SHARD_COUNT);
+          IntStream.range(0, toIntExact(committee_count)).boxed().collect(Collectors.toList());
+      long SHARD_COUNT = Constants.SHARD_COUNT;
       for (Integer committee_offset : comnmittee_range) {
-        UnsignedLong offset = UnsignedLong.valueOf(committee_offset);
-        UnsignedLong shard = state.getCurrent_shuffling_start_shard().plus(offset).mod(SHARD_COUNT);
-        UnsignedLong crosslink_epoch =
-            state.getLatest_crosslinks().get(shard.intValue()).getEpoch();
-        if (crosslink_epoch.compareTo(validator_registry_update_epoch) > 0) {
+        long shard = state.getCurrent_shuffling_start_shard() + committee_offset % SHARD_COUNT;
```

Comment:
The OOO is incorrect here. Should be `long shard = (state.getCurrent_shuffling_start_shard() + committee_offset) % SHARD_COUNT;`